zebROS_ws/src/behaviors/src/game_piece_path_gen.cpp

```diff
@@ -0,0 +1,137 @@
+#include <ros/ros.h>
+#include <cmath>
+#include <std_srvs/Trigger.h>
+#include <algorithm>
+#include "base_trajectory_msgs/GenerateSpline.h"
+#include "behavior_actions/DynamicPath.h"
+#include "behavior_actions/GamePiecePickup.h"
+#include "field_obj/Detection.h"
+#include "trajectory_msgs/JointTrajectoryPoint.h"
+
+// Create service clients
+ros::ServiceClient spline_gen_cli;
+ros::ServiceClient dynamic_path_cli;
+
+field_obj::Detection lastObjectDetection;
+
+// Callback function to retrieve the most recent object detection message
+void objectDetectCallback(field_obj::DetectionConstPtr msg)
+{
+	lastObjectDetection = *msg;
+}
+
+trajectory_msgs::JointTrajectoryPoint generateTrajectoryPoint(double x, double y, double rotation) {
+	trajectory_msgs::JointTrajectoryPoint point;
+	point.positions.resize(3);
+	point.positions[2] = rotation;
+	point.positions[0] = x;
+	point.positions[1] = y;
+	return point;
+}
+
+bool genPath(behavior_actions::GamePiecePickup::Request& req, behavior_actions::GamePiecePickup::Response& res)
+{
+	res.success = false; // Default to failure
+  base_trajectory_msgs::GenerateSpline spline_gen_srv;
+
+  spline_gen_srv.request.header = lastObjectDetection.header;
+
+  const size_t objects_num = std::min((int)(lastObjectDetection.objects.size()), (int)(req.max_objects));
+	ROS_INFO_STREAM("game_piece_path_gen : objects_num: " << objects_num);
+
+  std::vector<std::array<double, 3>> points;
+  points.push_back({0, 0, 0}); // may need to worry about transforms between beginning/end point and intake?
```

Comment:
As we talked about, the transforms seem to be working fine, so you can remove this comment (good to have it in there for tracking, but now that it is resolved remove it so others won't hit it in the future and wonder about how relevant it is).

Reply:
I think that was a comment left behind from the code before I started editing it, I didn't know much about transforms when I started working on this :) I just deleted it.